src/theory/sets/normal_form.h

```diff
@@ -45,10 +45,11 @@ class NormalForm {
     else
     {
       ElementsIterator it = elements.begin();
-      Node cur = nm->mkNode(kind::SINGLETON, *it);
+      Node cur = nm->mkSingleton(it->getType(), *it);
```

Comment:
I think you should pass the type to use as an argument of this function.  It is definitely not safe to use getType here, e.g. think of set that mixes 1.0 and 1.5.

Reply:
Done. The function has `TypeNode setType` which we can use to retrieve the element type.